bundles/org.openhab.io.openhabcloud/src/main/java/org/openhab/io/openhabcloud/internal/CloudClient.java

```diff
@@ -157,159 +163,170 @@ public CloudClient(HttpClient httpClient, String uuid, String secret, String bas
-            public void call(Object... args) {
-                logger.debug("Socket.IO connected");
-                isConnected = true;
-                onConnect();
-            }
-        }).on(Socket.EVENT_CONNECTING, new Emitter.Listener() {
-            @Override
-            public void call(Object... args) {
-                logger.debug("Socket.IO connecting");
-            }
-        }).on(Socket.EVENT_RECONNECTING, new Emitter.Listener() {
-            @Override
-            public void call(Object... args) {
-                logger.debug("Socket.IO re-connecting (attempt {})", args[0]);
-            }
-        }).on(Socket.EVENT_RECONNECT, new Emitter.Listener() {
-            @Override
-            public void call(Object... args) {
-                logger.debug("Socket.IO re-connected successfully (attempt {})", args[0]);
-            }
-        }).on(Socket.EVENT_RECONNECT_ERROR, new Emitter.Listener() {
-            @Override
-            public void call(Object... args) {
-                if (args[0] instanceof Exception) {
-                    Exception e = (Exception) args[0];
-                    logger.debug("Socket.IO re-connect attempt error: {} {}", e.getClass().getSimpleName(),
-                            e.getMessage());
-                } else {
-                    logger.debug("Socket.IO re-connect attempt error: {}", args[0]);
-                }
-            }
-        }).on(Socket.EVENT_RECONNECT_FAILED, new Emitter.Listener() {
-            @Override
-            public void call(Object... args) {
-                logger.debug("Socket.IO re-connect attempts failed. Stopping reconnection.");
-            }
-        }).on(Socket.EVENT_DISCONNECT, new Emitter.Listener() {
-            @Override
-            public void call(Object... args) {
-                if (args.length > 0) {
-                    logger.debug("Socket.IO disconnected: {}", args[0]);
-                } else {
-                    logger.debug("Socket.IO disconnected");
-                }
-                isConnected = false;
-                onDisconnect();
-            }
-        }).on(Socket.EVENT_ERROR, new Emitter.Listener() {
-            @Override
-            public void call(Object... args) {
-                if (CloudClient.this.socket.connected()) {
-                    if (logger.isDebugEnabled() && args.length > 0) {
-                        logger.error("Error during communication: {}", args[0]);
+                })//
+                .on(Socket.EVENT_RECONNECT_FAILED,
+                        args -> logger.debug("Socket.IO re-connect attempts failed. Stopping reconnection."))//
+                .on(Socket.EVENT_DISCONNECT, args -> {
+                    if (args.length > 0) {
+                        logger.info("Socket.IO disconnected: {}", args[0]);
                     } else {
-                        logger.error("Error during communication");
+                        logger.info("Socket.IO disconnected");
```

Comment:
Shouldn't these be debug (recoverable error) or warn?

Reply:
Made it warn in https://github.com/openhab/openhab-addons/pull/12430/commits/49cec9cf3d16454d64c14bbfde5e879e2fb7bd01

I think it's better that by default users see when disconnects happen